### DISCUSSION TITLE
Resetting window position /  width / height on Linux does not work

### DISCUSSION THREAD

**Discussion by llfab** [AvaloniaUI/Avalonia#9341] [Q&A] [answered]
I am animating my MainView (Window) in the following way:
- Animating from Opacity 1.0 to 0.0
- Resetting Window properties like Position, Width, Height, TopMost, ShowInTaskbar
- Animating from Opacity 0.0 to 1.0

Even though the only working way I found was some code behind in MainWindow this works on Windows (see video).

On Linux this does not work. The Window animates to 0 Opacity and back. Also the content is correctly changed, but the Window still keeps its Position, Width and Height. 

```
        private void AnimateToApplicationModeNormal()
        {
            using (var disposables = new CompositeDisposable())
            {
                disposables.Add(this.SetValue(Visual.OpacityProperty, 0, BindingPriority.Animation)!);

                // run the hide animation
                _hideWindowDesktopAnimation.RunAsync(this, null, default);

                // start the timer to re-set window state and trigger show animation
                _toNormalTimer.Start();
            }
        }

        private void OnToNormalTimerTick(object sender, EventArgs e)
        {
            _toNormalTimer.Stop();
            UpdateToApplicationModeNormal();
            _showWindowNormalAnimation.RunAsync(this, null, default);
        }

        private void AnimatetoApplicationModeDesktop()
        {
            using (var disposables = new CompositeDisposable())
            {
                disposables.Add(this.SetValue(Visual.OpacityProperty, 0, BindingPriority.Animation)!);

                // run the hide animation
                _hideWindowNormalAnimation.RunAsync(this, null, default);

                // start the timer to re-set window state and trigger show animation
                _toDesktopTimer.Start();
            }
        }

        private void OnToDesktopTimerTick(object sender, EventArgs e)
        {
            _toDesktopTimer.Stop();
            UpdateToApplicationModeDesktop();
            _showWindowDesktopAnimation.RunAsync(this, null, default);
        }

        private void UpdateToApplicationModeNormal()
        {
            Width = _originalWidth;
            Height = _originalHeight;
            Position = _originalPosition;
            Topmost = false;
            ShowInTaskbar = true;
            Background = _normalModeBackground;

            MainWindowViewModel vm = DataContext as MainWindowViewModel;
            vm.IsNormalMode = true;
            vm.IsDesktopMode = false;

            InvalidateVisual();
        }

        private void UpdateToApplicationModeDesktop()
        {
            Width = _desktopModeWidth;
            Height = _desktopModeHeight;
            Position = _desktopModePosition;
            Topmost = _desktopModeTopmost;
            ShowInTaskbar = _desktopModeShowInTaskbar;
            Background = _desktopModeBackground;

            MainWindowViewModel vm = DataContext as MainWindowViewModel;
            vm.IsNormalMode = false;
            vm.IsDesktopMode = true;

            InvalidateVisual();
        }
```

The above timer is a dispatcher timer, so everything is in the correct thread.

Questions:
- How do I change Position, Width, Height for Linux?
- How could I nicely animate the entire change without the ugly timer but with a full animation (story board)?

**Comment by llfab** [comment]
https://user-images.githubusercontent.com/31031996/199558419-c3c8002c-131c-45da-bc27-c0a485cafa3e.mp4

**Reply by llfab** [reply]
So the feature I want is:
- let the current full Window disappear
- let (optimally the same) Window appear in a different position with different width / height and content.

**Reply by llfab** [reply]
Yes, I did this with a single MainWindow as I found that opening 2 different top level Windows with Avalonia would also not be so great. At least that may be an option that I have not investigated yet...

**Comment by llfab** [comment, marked as answer]
This is the Linux version...

https://user-images.githubusercontent.com/31031996/199560202-ddd19cfd-d816-457a-aa64-49c7cf4e5c00.mp4

**Reply by llfab** [reply]
There all animation and content handling is ok but the Window still has its previous position, width, height.

**Reply by kekekeks** [reply]
The window looks maximized. Not sure if position changes can take effect if that's the case.

**Reply by llfab** [reply]
At least from a settings perspective it is not but due to its size it may have been assigned with that. I will check. 

Anyhow, would you propose to rather use 2 different Windows and exchange their visibility or reset the one and only. The experience with Qt/Qml on different platforms / graphic adapters has shown that interacting as little as possible with the OS windowing system is a suitable principle. Unclear, which of the 2 options is more "little".

**Reply by llfab** [reply]
So, if I reduce window size to not fill the entire screen, relocation kind of works but still not fully correct to its defined position. Hence, maybe two separate Windows without settings change (but just visibility switch) may be better....

**Comment by llfab** [comment]
According to my tests with Ubuntu the Position, Width, Height must be set to a window before the first call of .Show(). On Windows this seems not the case. Anyhow, the testing proved that it is better to work with two separate Windows that show and hide.

Thanks for that!
